### ISSUE TITLE
Support math equation numbering and its corresponding reference

### ISSUE THREAD

**Issue by facat** [typora/typora-issues#48]

**Comment by azag0** [comment]
This would be indeed a great feature to have.

**Comment by abnerlee** [comment]
Is there any markdown syntax to support this feature?

**Comment by azag0** [comment]
This should all be handled by [mathjax](http://mathjax.readthedocs.org/en/latest/tex.html#automatic-equation-numbering), but it does not seem to work in Typora. I'm not sure how Typora calls mathjax, but it might be that mathjax is called separately for each equation and hence it doesn't know about the references. To sum up, I imagine this would work in Typora:

```markdown
$$
\begin{equation}
a^2+b^2=c^2
\label{eq:pythagoras}
\end{equation}
$$

Referring to Eq. $\eqref{eq:pythagoras}$.
```

**Comment by azag0** [comment]
Actually, it does seem to work when pasted all at once in a fresh new document.

![screen shot 2016-02-04 at 20 31 25](https://cloud.githubusercontent.com/assets/940353/12827081/4cd07f14-cb7e-11e5-917d-226720db5445.png)

But after editing (while changing nothing), the math block stays unprocessed,

![screen shot 2016-02-04 at 20 32 01](https://cloud.githubusercontent.com/assets/940353/12827113/6f0a1306-cb7e-11e5-96fd-5684ccfa913b.png)

There might be some useful information [here](http://meta.math.stackexchange.com/questions/11392/mathjax-preview-broken-when-equations-contain-labels).

**Comment by abnerlee** [comment]
Thanks, I will check the doc

**Comment by RomainBui** [comment]
I would like to second here, this would be a great future.

**Comment by ropeonfire** [comment]
I would love to have this feature -- I've attempted to modify the MathJax.Hub.Config to include equationNumbers (accessed by toggling the DevTools):
```TeX: { extensions: ["noUndefined.js", "autoload-all.js", "AMSmath.js", "AMSsymbols.js"],
equationNumbers: { autoNumber: "AMS" } }```
This seems to have no effect - and I'm not even sure it's supposed to since I know nothing of html, etc...

**Comment by ropeonfire** [comment]
I've modified the MathJax.Hub.Config in "...\Typora\resources\app\window.html" to include auto-numbering, as shown in the snippet in my comment above. This allows the use of the `\label` syntax; the equations are auto-numbered in the order they are written/processed, as expected. Also, the "multiply defined" error will return if the equation is edited after it is initially rendered. If this happens, one can get things re-rendered and re-numbered appropriately by closing/opening the file. 

Remaining issues: 
- In the PDF export, references generated by `\eqref{}` or `\ref{}` appear to be clickable but don't actually seem to be associated and do not jump to the referenced equations. OTH, the references in the .html export seem to work as expected...Wish it were the other way around... :P
- It may be possible to use MathJax.Hub.Queue (see [the relevant MathJax docs here](http://docs.mathjax.org/en/latest/advanced/typeset.html#reset-equation-numbers)). Unfortunately I am not familiar with where this needs to be included...I've tried adding it to "window.html" with poor results. I might just be putting it in the wrong place (I'm just blindly plugging away here). **Any ideas?**

**Comment by truefalsename** [comment]
On a mac you can turn on equation autonumbering by editing /Applications/Typora.app/Contents/Resources/TypeMark/lib/MathJax-2.6.1/jax/input/TeX/config.js

changing "none" to "AMS" or "all" in equationNumbers:{autoNumber:"none",formatNumber:function(a){return a},formatTag:function(a){return"("+a+")"}.

**Comment by ickc** [comment]
I configured the MathJax in `/Applications/Typora.app/Contents/Resources/TypeMark/index.html` and it would work, except that the numbering could have problems.

It seems typora uses the MathJax to rendered it once and stored the resulted CHTML (or HTML+CSS?). The problem then is if I edit the document non-linearly, the numbering will be incorrect. In fact, when I tested it, even if I have only 1 numbered equation, editing the same equation repeatedly will results in incrementing the number count on that equation.

Given how typora works (live view on the source), this behavior on the preview is acceptable. However, when it is exported as HTML, the results would still be the same. So I suggest it to do a complete rendering at the time of export to yield a correct result.
